ISICkonto/Support/Result.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  Result.swift
+//  ISICkonto
+//
+//  Created by Vendula Švastalová on 13/03/2019.
+//  Copyright © 2019 Vendula Švastalová. All rights reserved.
+//
+
+import Foundation
+
+public enum ResultInformation {
+    case inProgress, success, failure, loggedIn
+}
+
+public struct Result<Element> {
+    public let info: ResultInformation
+    public let message: String?
+    public let element: Element?
+    
+    public init(info: ResultInformation, message: String?, element: Element?) {
```

Comment:
Když použiješ `struct`, tak máš tento init generovaný, tedy není potřeba ho explicitně psát

Reply:
aha, to jsem nevedela, takze struct s sebou nese nejakej defaultni init kterej inicializuje vsechny properties? kdyz bych chtela jinej init tak ten si dodelam ale tento tam vzdy proste je?

Reply:
`struct` má na sobě defaultně přesně ten init, co tam máš napsaný. Tedy z toho initu jsi schopná nastavit všechny parametry `struct`y. Pokud vytvoříš přímo v definici `struct`y vlastní init, tak o ten generovaný přijdeš. Best practice je vlastní inity dělat v extensioně

Reply:
jasne, diky 👍